src/main/java/org/jenkinsci/plugins/workflow/actions/ErrorAction.java

```diff
@@ -75,8 +75,8 @@ public ErrorAction(@NonNull Throwable error) {
      * Some exceptions don't serialize properly. If so, we need to replace that with
      * an equivalent that captures the same details but serializes nicely.
      */
-    private boolean isUnserializableException(@CheckForNull Throwable error) {
-        if (error == null) {
+    private boolean isUnserializableException(@CheckForNull Throwable error, Set<Throwable> visited) {
+        if (error == null || !visited.add(error)) {
             return false;
```

Comment:
I considered returning `true` when `!visited.add(error)`, but then things would be 100% broken for all cyclic errors until https://github.com/jenkinsci/remoting/pull/645 is picked up. Right now, with this PR, if a cyclic error is serializable, I think everything works, even with the cycle.